### PR TITLE
chore(server): update global Node polyfills

### PR DESCRIPTION
**Pull request by kitten**
# Why

This updates global Node polyfills to use `@remix-run/node@^2.7.2` instead, since it drops a couple of polyfills in favour of using Node internals.

# How

- Updated to `@remix-run/node@^2.7.2`
- Removed `ExpoRequest` and `ExpoResponse` globals and dropped `ExpoURL` in favour of built-ins
- Added `Response.{json,error,redirect}` statics, which seem to be missing in `@remix-run/web-fetch`

The globals have been removed since they're not necessary, this also fixes:
- `instanceof Response` and `instanceof Request` checks for third-party libraries
- Potentially, issues with Bun where previously the globals overrides led to issues (no crash seems to occur as far as I can tell)

# Test Plan

- Tested manually against our [Express](https://docs.expo.dev/router/reference/api-routes/#express) instructions on Node 20
  - Same instructions were tested with bun (via `bun run express.js`)
- Tested manually against our [Netlify](https://docs.expo.dev/router/reference/api-routes/#express) instructions on Node 20, running locally in `netlify dev`
- ~~Tested manually against our [Vercel](https://docs.expo.dev/router/reference/api-routes/#vercel) instructions on Node 20, running locally in `vercel dev`~~
  - This currently isn't working for me due to some local setup issues 😿 `vercel dev` is failing for unrelated reasons.
- Tested API routes in `apps/router-e2e/__e2e__/server/app/api` locally via `expo start`

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by amandeepmittal — APPROVED** [review]

File: apps/router-e2e/__e2e__/server/app/api/[dynamic]+api.ts
Comment (EvanBacon) [inline apps/router-e2e/__e2e__/server/app/api/[dynamic]+api.ts:3]: Isn't Response.json on the type? https://developer.mozilla.org/en-US/docs/Web/API/Response/json_static

File: packages/@expo/cli/CHANGELOG.md
Comment (EvanBacon) [inline packages/@expo/cli/CHANGELOG.md:44]:
```suggestion
- Update to remove `ExpoRequest/ExpoResponse` imports from `@expo/server`. ([#27261](https://github.com/expo/expo/pull/27261) by [@kitten](https://github.com/kitten))
```

File: packages/@expo/server/src/index.ts
Comment (EvanBacon) [inline packages/@expo/server/src/index.ts:109]: Also delete the code for populating expoUrl via `expoUrl.searchParams`

File: apps/router-e2e/__e2e__/server/app/api/[dynamic]+api.ts
Comment (kitten) [reply apps/router-e2e/__e2e__/server/app/api/[dynamic]+api.ts:3]: Depending on the TypeScript version and used types this won't always be available, and we can't “force” it to be there, since we can't re-declare `var Response` 😅 
In our case it's missing because we're on an older TypeScript version

**Comment by kitten** [comment]
@EvanBacon: `ExpoRequest` and `ExpoResponse` are re-added now as deprecated globals, and the remaining `ExpoURL` remnants should have been removed. Next up would be double-checking Vercel and then this is good to go, unless we find anything else to fix up before merging ❤️

File: apps/router-e2e/__e2e__/server/app/api/[dynamic]+api.ts
Comment (byCedric) [reply apps/router-e2e/__e2e__/server/app/api/[dynamic]+api.ts:3]: Can we do something like this for `.json` as well? https://github.com/expo/expo/pull/27261/files#diff-4ab881a1080ac3bedcac4050238c0cb055661849b325b897c27ebf16bf3ee192R45-R62

File: packages/@expo/server/src/environment.ts
Comment (EvanBacon) [inline packages/@expo/server/src/environment.ts:34]: Similar to redirect, we should have a test for this behavior.

**Review by EvanBacon — APPROVED** [review]
nice work

**Pull request merged**
